### PR TITLE
Rename `tools/Makefile.*` to have the `.mk` extension

### PR DESCRIPTION
**Pull request by AlanRosenthal**
## Summary
This PR renames `tools/Makefile.*` to have the `.mk` extension. This PR also updates `README.md` and other files that references the other files.
Note: Skipped Makefile.host for this PR since it caused failures in CI for the sim build

## Impact
By using a standard extension for Makefiles (https://www.file-extension.info/format/mk), editors will auto-format files.
This change will also improve developer ergonomics when searching for specific files

## Testing
Verified locally that build still works, CI will verify more!

### CODE REVIEW

**Comment by Ouss4** [comment]
@AlanRosenthal we've got sone chicken'n'egg conflicts here. :)
Now that https://github.com/apache/incubator-nuttx/pull/5147 is merged, this PR has conflicts.  It will also have the another conflict when https://github.com/apache/incubator-nuttx/pull/5162 is merged.

File: boards/mips/pic32mz/pic32mz-starterkit/README.txt [hunk 1]
```diff
@@ -78,9 +78,9 @@ On Board Debug Support
   The starter kit includes a PIC24FJ256GB106 USB microcontroller that
   provides debugger connectivity over USB. The PIC24FJ256GB106 is hard-wired
   to the PIC32 device to provide protocol translation through the I/O pins
-  of the PIC24FJ256GB106 to the ICSP� pins of the PIC32 device.
+  of the PIC24FJ256GB106 to the ICSP� pins of the PIC32 device.
```
Comment (xiaoxiang781216) [inline boards/mips/pic32mz/pic32mz-starterkit/README.txt:81]: revert the change?

File: boards/mips/pic32mz/pic32mz-starterkit/README.txt [hunk 2]
```diff
@@ -78,9 +78,9 @@ On Board Debug Support
...
 
-  If MPLAB� REAL ICE� or MPLAB ICD 3 is used with the starter kit,
+  If MPLAB� REAL ICE� or MPLAB ICD 3 is used with the starter kit,
```
Comment (xiaoxiang781216) [inline boards/mips/pic32mz/pic32mz-starterkit/README.txt:83]: ditto

File: boards/mips/pic32mx/pic32mx-starterkit/README.txt [hunk 3]
```diff
@@ -1038,7 +1038,7 @@ Where <subdir> is one of the following:
        as NSH built-in built in functions.
 
        To use USB device, connect the starter kit to the host using a cable
-       with a Type-B micro-plug to the starter kit�s micro-A/B port J5, located
+       with a Type-B micro-plug to the starter kit�s micro-A/B port J5, located
```
Comment (xiaoxiang781216) [inline boards/mips/pic32mx/pic32mx-starterkit/README.txt:1041]: revert the change

File: boards/mips/pic32mx/pic32mx-starterkit/README.txt [hunk 4]
```diff
@@ -522,7 +522,7 @@ On Board Debug Support
   that provides debugger connectivity over USB. The PIC32MX440F512H is hard-wired
   to the PIC32 device to provide two types of protocol translation:
 
-    - I/O pins of PIC32MX440F512H to the ICSP� pins of the PIC32
+    - I/O pins of PIC32MX440F512H to the ICSP� pins of the PIC32
```
Comment (xiaoxiang781216) [inline boards/mips/pic32mx/pic32mx-starterkit/README.txt:525]: revert

File: boards/mips/pic32mz/pic32mz-starterkit/README.txt (hunk 2 above)
Comment (AlanRosenthal) [reply boards/mips/pic32mz/pic32mz-starterkit/README.txt:83]: weird, not sure what happened here. will revert

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by Ouss4 — APPROVED** [review]
LGTM!
